utils/profdata_merge_worker.py

```diff
@@ -0,0 +1,218 @@
+            self.report("Adding %s to filename_buffer." % filename)
+            if len(self.filename_buffer) >= 10:
+                self.merge_file_buffer()
+                FILE_QUEUE.task_done()
+
+def run_server():
+    pid = os.getpid()
+    if os.path.exists(CONFIG.pid_file_path):
+        with open(CONFIG.pid_file_path) as pidfile:
+            pid = pidfile.read()
+            printsync(("existing process found with pid %s." +
+                       "Ensure there are no other test runners running," +
+                       "and delete the file at %s")
+                       % (pid, CONFIG.pid_file_path))
+        return
+
+    with open(CONFIG.pid_file_path, "w") as pidfile:
+        pidfile.write(str(pid))
+
+    processes = [ProfdataMergerProcess() for _ in range(10)]
+    for p in processes:
+        p.start()
+
+    server = SocketServer.TCPServer(SERVER_ADDRESS, ProfdataTCPHandler)
+    server.serve_forever()
+
+    for p in processes:
+        # force each merge worker to gracefully exit
+        FILE_QUEUE.put(None)
+
+    for p in processes:
+        printsync("waiting for %s to finish..." % p.name)
+        p.join()
+
+    # now that all workers have completed, merge all their files
+    merge_final = ProfdataMergerProcess()
+    merge_final.profdata_path = CONFIG.final_profdata_path
+    for p in processes:
+        if os.path.exists(p.profdata_path):
+            printsync("merging " + p.profdata_path + "...")
+            merge_final.filename_buffer.append(p.profdata_path)
+    merge_final.merge_file_buffer()
+
+def stop_server(args):
+    sock = socket.socket(socket.AF_INET, socket.SOCK_STREAM)
+    sock.connect(SERVER_ADDRESS)
+    sock.send(TESTS_FINISHED_SENTINEL)
+    sock.close()
+
+def start_server(args):
+    global CONFIG
+    CONFIG = Config(args.debug, args.output_dir, args.no_remove)
+    if not CONFIG.debug:
+        pid = os.fork()
+        if pid != 0:
+            sys.exit(0) # kill the parent process we forked from.
+    try:
+        run_server()
+    finally:
+        if os.path.exists(CONFIG.pid_file_path):
+            os.remove(CONFIG.pid_file_path)
+        if os.path.exists(CONFIG.tmp_dir):
+            shutil.rmtree(CONFIG.tmp_dir, ignore_errors=True)
+
+if __name__ == "__main__":
+    parser = argparse.ArgumentParser()
+
+    subparsers = parser.add_subparsers()
+
+    start = subparsers.add_parser("start")
+    start.add_argument("-d", "--debug",
+        help="Run in foreground and report status.",
+        action="store_true")
+    start.add_argument("-o", "--output-dir",
+        help="The directory to write the PID file and final profdata file.",
+        default="/tmp")
+    start.add_argument("--no-remove",
+        action="store_true",
+        help="Don't remove profraw files after merging them.")
+    start.set_defaults(func=start_server)
+
+    stop = subparsers.add_parser("stop")
+    stop.set_defaults(func=stop_server)
+
+    args = parser.parse_args()
+    args.func(args)
```

Comment:
It would be nice to create a Python module for this, as opposed to a single file. It'd allow you to:

1. Add tests for the module. You could also add the tests to the Swift validation test suite, as in #810.
2. Split the file up. `Config`, `ProfdataMergerProcess`, `ProfdataTCPHandler`, and even the argument parser could each get their own file. Where possible, it would be nice to unit test these classes.